designer/client/conditions/inline-condition-values.js

```diff
@@ -0,0 +1,49 @@
+const conditionValueFactories = {}
+
+export class ConditionValue {
```

Comment:
I suppose this is an abstract class, without being an abstract class? It seems "wrong" in vanilla ECMAScript, maybe we should implement flow/typescript to better support this.. big job.. put it in the backlog..

Reply:
Exactly, it's not ideal. I did think about the whole typescript thing but figured way too much work for this PR!

Reply:
I have started a discussion on the maintainers board